src/Orleans.Core/Serialization/SerializationManager.cs

```diff
@@ -222,6 +222,22 @@ public void Register(Type t, DeepCopier cop, Serializer ser, Deserializer deser)
             Register(t, cop, ser, deser, false);
         }
 
+        private object InitializeSerializer(Type serializerType)
+        {
+            if (this.ServiceProvider == null)
+                return null;
+
+            if (!serializerType.GetCustomAttributes(typeof(SerializerAttribute), true).Any())
+                return null;
+
+            var constructors = serializerType.GetConstructors();
+            if (constructors == null || constructors.Length < 1)
+                return null;
+
+            var serializer = ActivatorUtilities.CreateInstance(this.ServiceProvider, serializerType);
```

Comment:
Maybe it's out of scope, but does it make sense to use GetServiceOrCreateInstance here instead since we expect 1 instance of each serializer?

Reply:
To uses that, we would need to register the serializers in the `IServiceProvider`, correct?

I am not sure we want to register all of them, since you may have A LOT of serializers in your app

Reply:
No, this means that *if* it is registered as a service, it will use that, otherwise falls back to CreateInstance.

Reply:
BTW, are you doing this change or decided not to?